ksqldb-execution/src/main/java/io/confluent/ksql/execution/interpreter/terms/BasicTerms.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.interpreter.terms;
+
+import io.confluent.ksql.execution.interpreter.TermEvaluationContext;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.schema.ksql.types.SqlTypes;
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+
+@SuppressWarnings("checkstyle:ClassDataAbstractionCoupling")
+public final class BasicTerms {
```

Comment:
```suggestion
public final class LiteralTerms {
```

I think this describes the class a little better.

Reply:
Done.  I moved all of the interfaces to a class called `TypedTerms`.